pkg/controller/certificaterequest/issue_certificate.go

```diff
@@ -57,6 +58,7 @@ func (r *ReconcileCertificateRequest) IssueCertificate(reqLogger logr.Logger, cr
 	if proceed {
 		reqLogger.Info("write permissions for DNS has been validated")
 	} else {
+		err = errors.New("failed to get write access to DNS record")
```

Comment:
why not use the err returned by `dnsClient.ValidateDNSWriteAccess(reqLogger, cr)` ?

Reply:
In that case, the error is nil per the previous if loop, however, we logically handled it as an error hence creating a new error to align the behaviour